src/engine/enginebuffer.cpp

```diff
@@ -1164,8 +1164,12 @@ void EngineBuffer::processSeek(bool paused) {
     }
 
     if (!paused && (seekType & SEEK_PHASE)) {
-        position = m_pBpmControl->getNearestPositionInPhase(position, true, true);
+        double syncPosition = position;
+        double requestedPosition = position;
+        syncPosition = m_pBpmControl->getNearestPositionInPhase(position, true, true);
+        position = m_pLoopingControl->getSyncPositionInsideLoop(requestedPosition, syncPosition);
     }
+
     if (position != m_filepos_play) {
         setNewPlaypos(position, adjustingPhase);
```

Comment:
i think we can now get rid of the adjustingPhase flag, because the position is always inside the loop now.